test/client.tests.js

```diff
@@ -0,0 +1,49 @@
+const { assert } = require('chai');
+const { get: getConfig } = require('../lib/config');
+const { get: getClient } = require('../lib/client');
+const nock = require('nock');
+const pkg = require('../package.json');
+
+describe('client initialization', function() {
+  describe('default case', function() {
+    const config = getConfig({
+      clientID: '__test_client_id__',
+      clientSecret: '__test_client_secret__',
+      issuerBaseURL: 'https://flosser.auth0.com',
+      baseURL: 'https://theapplication.com',
+    });
+
+    let client;
+    before(async function() {
+      client = await getClient(config);
+
+      nock('https://flosser.auth0.com')
```

Comment:
Nock was the best way I could think of/find for getting the options being used. They are not (understandably) exposed outside of the library.